src/lib/openapi/spec/create-user-schema.ts

```diff
@@ -5,24 +5,52 @@ export const createUserSchema = {
     type: 'object',
     additionalProperties: false,
     required: ['rootRole'],
+    description:
+        'The payload must contain at least one of the name and email properties, though which one is up to you. For the user to be able to log in to the system, the user must have an email.',
     properties: {
         username: {
+            description:
+                "The user's username. Must be provided if email is not provided.",
             type: 'string',
+            example: 'hunter',
         },
         email: {
+            description:
+                "The user's email address. Must be provided if username is not provided.",
             type: 'string',
+            example: 'user@example.com',
         },
         name: {
+            description: "The user's name (not the user's username).",
             type: 'string',
+            example: 'Sam Seawright',
         },
         password: {
             type: 'string',
+            example: 'k!5As3HquUrQ',
+            description: 'Password for the user',
         },
         rootRole: {
-            type: 'number',
+            description:
+                "The role to assign to the user. Can be either the role's ID or its unique name.",
+            oneOf: [
```

Comment:
oneOf with integer or string + coercion: true in our openapi lib causes rootRole to always be a string. We need to compensate for it in the controller by coercing back